src/poggit/ci/builder/DefaultProjectBuilder.php

```diff
@@ -195,6 +199,65 @@ protected function build(Phar $phar, RepoZipball $zipball, WebhookProjectModel $
         return $result;
     }
 
+    private function runDynamicCommandList(string $pharPath, string $pluginName, int $buildId){
+        if($pluginName === null){
+            Meta::getLog()->e("Failed to start dyncmdlist, no plugin name found.");
+            return;
+        }
+
+        $id = "dyncmdlist-" . substr(Meta::getRequestId() ?? bin2hex(random_bytes(8)), 0, 4) . "-" . bin2hex(random_bytes(4));
+
+        Meta::getLog()->v("Starting dyncmdlist flow with ID '{$id}'");
+
+        Lang::myShellExec("docker run --name {$id} --cpus=1 --memory=256M -v ${pharPath}:/input/plugin.phar pmmp/dyncmdlist:0.1.1 ./wrapper.sh ${pluginName}", $stdout, $stderr, $exitCode);
+
+        if($exitCode !== 0){
+            Meta::getLog()->e("dyncmdlist failed with code '{$exitCode}', stdout: {$stdout}, stderr: {$stderr}");
+            return;
+        }
+
+        $result = json_decode($stdout, true);
+
+        if($result === null || !isset($result["status"])){
+            Meta::getLog()->e("Failed to parse results from dyncmdlist, stdout: {$stdout}, stderr: {$stderr}");
+            return;
+        }
+
+        if($result["status"] === false){
+            Meta::getLog()->e("dyncmdlist failed with error '{$result["error"]}'");
+            return;
+        }
+
+        if(sizeof($result["commands"]) === 0) {
+            Meta::getLog()->v("dyncmdlist found no commands.");
+            return;
```

Comment:
are you sure we should return here?

Reply:
Don’t have to but wouldn’t do anything after 🤷‍♂️